admin/class-primary-term-admin.php

```diff
@@ -252,7 +253,44 @@ protected function get_mapped_taxonomies_for_js( $taxonomies ) {
 	}
 
 	/**
-	 * Returns an array suitable for use in the javascript.
+	 * Gets a list of taxonomies based on the passed taxonomy and checks for ancestor taxonomies.
+	 *
+	 * @param string $taxonomy The taxonomy to get the list for.
+	 *
+	 * @return array The list of taxonomies.
+	 */
+	protected function get_taxonomies( $taxonomy ) {
+		$terms = get_terms( $taxonomy, array( 'hide_empty' => false ) );
```

Comment:
If you pass the value of `$terms` as a argument to this method it will become more testable. This also separates our logic from the WordPress logic.